festim/generic_simulation.py

```diff
@@ -206,6 +206,12 @@ def attribute_source_terms(self):
 
         # set sources
         for source in self.sources:
+            if source.field == "T" and not isinstance(
+                self.T, festim.temperature.temperature_solver.HeatTransferProblem
+            ):  # check that there is not a source defined in T as the same time as a festim.Temperature
+                raise ValueError(
```

Comment:
Maybe this should be a `TypeError` instead since the issue is that the temperature doesn't have the right type?

```suggestion
                raise TypeError(
```